### PR TITLE
Removed exception catching and error reporting in GetRunnerFor, since…

### PR DESCRIPTION
**Pull request by OsirisTerje**
… both calling methods have the same. #302 

The GetRunnerFor method is called from the discoverer and the executor and both calls are within try/catch blocks.  We should be able to safely remove it from the GetRunnerFor method, and keep it in the calling methods.
Further, as the #302 issue points out, it should not be an error if it is a BadImageException.  This is correctly handled in the calling method.

### CODE REVIEW

File: src/NUnitTestAdapter/NUnitTestAdapter.cs [hunk 1]
```diff
@@ -136,17 +136,7 @@ protected void Initialize(IDiscoveryContext context, IMessageLogger messageLogge
...
-            {
-                return TestEngine.GetRunner(package);
-            }
-            catch (Exception ex)
-            {
-                TestLog.Error("Error: Unable to get runner for this assembly. Check installation, including any extensions.");
-                TestLog.Error(ex.GetType().Name + ": " + ex.Message);
-                throw;
-            }
+            return TestEngine.GetRunner(package);
```
Comment (CharliePoole) [inline src/NUnitTestAdapter/NUnitTestAdapter.cs]: It's easy to get this kind of double exception handling if we don't have a test. Maybe we should add a test that shows this method __does__ throw the exception in this situation.

**Comment by CharliePoole** [comment]
Seems like you have  an odd dependency here. Shouldn't it be depending on the version you are building, not on a package?

**Comment by OsirisTerje** [comment]
The former failure was missing adapter for the demo project, where it is added, as a package. 
Current failure seems to be downloading nuget packages, but hard to see what is wrong here.  
@CharliePoole @rprouse :  Any ideas?

**Comment by rprouse** [comment]
Initial pass through the build output, the adapter tests successfully builds all dependent projects, but then fails itself. It looks like none of its references are being pulled in since it isn't finding the VS SDK namespaces. Still looking...

**Comment by rprouse** [comment]
The build failures seem to be totally unrelated to your changes. It is failing to build the test project, not the demo which you changed. The first error I see is,

```
AdapterSettingsTests.cs(6,17): error CS0234: The type or namespace name 'VisualStudio' does not exist in the namespace 'Microsoft' (are you missing an assembly reference?) [C:\projects\nunit3-vs-adapter\src\NUnitTestAdapterTests\NUnit.TestAdapter.Tests.csproj]
```

I might be missing something in that wall of text, but it looks like the adapter built successfully. I checked the `appveyor.yml` file and it specifies Visual Studio 2017. I checked the list of environments and there isn't anything newer.

I have no idea. I am going to kick off another build. I might also try to rebuild master.

**Comment by rprouse** [comment]
We haven't had a successful build since the release, so it could be any number of changes causing the failure. Does it build from the command line for you if you delete your packages directory?

**Comment by OsirisTerje** [comment]
@rprouse Yes, removed local packages, and it still builds fine.  It does say that the .net  sdk  1.0.1 is already installed though.  Is that the case on the build server?

**Comment by CharliePoole** [comment]
@OsirisTerje This old issue just popped back up in a search related to a problem I'm working on. Mono.Cecil, by design, aims to read managed assemblies. As noted it throws an exception otherwise.

We would probably benefit by examining unmanaged assemblies and returning a flag that says they are unmanaged rather than trying to find a driver for it. Just wanted to get this notion on your radar. If the adapter could easily tell that the assembly is unmanaged, it could simply ignore it. As soon as we let Mono.Cecil get called, the trouble arises.

**Pull request merged**